docs/design-ux/search.md

```diff
@@ -0,0 +1,35 @@
+# Search
+
++ Filename search -- `fzf` search via path (ctrl + l)
++ Full lockbook content search (ctrl + shift + f)
++ Text editor search (ctrl + f)
+
+# Filename search
+
+You hand core a string, core will hand you back a list of paths that this string matches. Can be integrated into
+quick-navigation of applications like it's done in the Linux application.
+
+# Lockbook Content Search
+
+Be able to quickly find files that match a certain criteria inside your lockbook. Not all the files in your Lockbook are
+guaranteed to be available on disk. And not every file type lends itself well to search-time scans.
+
+I propose we add a field to `FileMetadata` that contains `tags` or some other search term. Upon document writes these
```

Comment:
The issue is that not every file will be available on disk, but metadata will. I would be on board with considering an expanding search which first operates on these tags, expands to local files, fetches remote files if possible to complete a thorough search.

rough heuristics: searching tags should take less than a second, searching local files should take less than 10 seconds, and then upto a minute for the complete search.

we could be pretty communicative of what operation is currently being done.

Reply:
**Discussed over VC**

Heuristics for generating tags will need to be played with. Ideas include "tags = top n%" or "tags include a portion of each page."

We have no guarantee that this will work out for a better search experience, however it's good enough for a first pass. A future implementation may instead look like:

* Search only covers offline (locally stored) files
* "Streaming" search that takes the time to pull files down to search them and updates the results list with new hits over time